pkgs/applications/editors/emacs/sources.nix

```diff
@@ -36,52 +37,66 @@ let
   };
 in
 {
-  emacs28 = import ./generic.nix {
-    pname = "emacs";
+  emacs28 = let
     version = "28.2";
+    rev = version;
+    hash = "sha256-4oSLcUDR0MOEt53QOiZSVU8kPJ67GwugmBxdX3F15Ag=";
+  in import ./generic.nix {
+    pname = "emacs";
+    inherit version;
     variant = "mainline";
+
     src = fetchFromSavannah {
       repo = "emacs";
-      rev = "28.2";
-      hash = "sha256-4oSLcUDR0MOEt53QOiZSVU8kPJ67GwugmBxdX3F15Ag=";
+      inherit rev hash;
     };
 
-    meta = mainlineMeta;
+    meta = mainlineMeta version;
   };
 
-  emacs29 = import ./generic.nix {
+  emacs29 = let
+    version = "29.1";
+    rev = version;
+    hash = "sha256-3HDCwtOKvkXwSULf3W7YgTz4GV8zvYnh2RrL28qzGKg=";
+  in
+    import ./generic.nix {
     pname = "emacs";
-    version = "29.1-rc1";
+    inherit version;
     variant = "mainline";
+
     src = fetchFromSavannah {
       repo = "emacs";
-      rev = "29.1-rc1";
-      hash = "sha256-p0lBSKsHrFwYTqO5UVIF/PgiqwdhYQE4oUVcPtd+gsU=";
+      inherit rev hash;
     };
 
-    meta = mainlineMeta;
+    meta = mainlineMeta version;
   };
 
-  emacs-macport = import ./generic.nix {
+  emacs-macport = let
+    version = "29.1";
+    rev = "emacs-29.1-mac-10.0";
+    hash = "sha256-TE829qJdPjeOQ+kD0SfyO8d5YpJjBge/g+nScwj+XVU=";
+  in import ./generic.nix {
     pname = "emacs-mac";
-    version = "28.2";
+    inherit version;
     variant = "macport";
+
     src = fetchFromBitbucket {
       owner = "mituharu";
       repo = "emacs-mac";
-      rev = "emacs-28.2-mac-9.1";
-      hash = "sha256-Ne2jQ2nVLNiQmnkkOXVc5AkLVkTpm8pFC7VNY2gQjPE=";
+      inherit rev hash;
```

Comment:
Perhaps we could retain the emacs-28 version of this too.

Reply:
I will reorder this.